### PR TITLE
DOC/TST: Clarify Series.str.get supports passing hashable label

### LINKED ISSUE #47911: DOC/TST: pandas.Series.str.get with dict elements supports passing hashable label
### Pandas version checks

- [X] I have checked that the issue still exists on the latest versions of the docs on `main` [here](https://pandas.pydata.org/docs/dev/)

### Location of the documentation

https://pandas.pydata.org/docs/dev/reference/api/pandas.Series.str.get.html

### Documentation problem

The fact that `Series.str.get` also works with Non-Numerical `dict` types isn't clear in the documentation.

### Suggested fix for documentation

Parameters should be `int` or `str` and an example using a dictionary should be added.

### PR DESCRIPTION
**Pull request by xr-chen**
- [ ] closes #47911 (Replace xxxx with the Github issue number)
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

**Comment by BeRT2me** [comment]
NO! https://github.com/pandas-dev/pandas/issues/47911#issuecomment-1201676711

File: doc/source/whatsnew/v1.5.0.rst [hunk 1]
```diff
@@ -279,6 +279,7 @@ Other enhancements
 - :class:`Series` reducers (e.g. ``min``, ``max``, ``sum``, ``mean``) will now successfully operate when the dtype is numeric and ``numeric_only=True`` is provided; previously this would raise a ``NotImplementedError`` (:issue:`47500`)
 - :meth:`RangeIndex.union` now can return a :class:`RangeIndex` instead of a :class:`Int64Index` if the resulting values are equally spaced (:issue:`47557`, :issue:`43885`)
 - :meth:`DataFrame.compare` now accepts an argument ``result_names`` to allow the user to specify the result's names of both left and right DataFrame which are being compared. This is by default ``'self'`` and ``'other'`` (:issue:`44354`)
+- :meth:`Series.str.get` now raises when ``i`` is not a integer (:issue:`47911`)
```
Comment (phofl) [inline doc/source/whatsnew/v1.5.0.rst:282]: Could you specify the error?

**Review by phofl — COMMENTED** [review]
Small comment, otherwise Lgtm

i is positional, e.g. strings should raise

edit: pre commit is failing, you can check this locally

**Comment by BeRT2me** [comment]
i is only positional in the documentation... it's definitely not strictly positional in current functionality.
https://github.com/pandas-dev/pandas/blob/14de3fd9ca4178bfce5dd681fa5d0925e057c04d/pandas/core/strings/object_array.py#L249

**Comment by mroeschke** [comment]
I am starting to agree with @BeRT2me, I don't think we can change this outright. 

I think the fact that `str.get` works for `object` series was just poorly document as there were past issues explaining some usage of this functionality. e.g. https://github.com/pandas-dev/pandas/issues/20671

**Comment by phofl** [comment]
Do you want to deprecate or add this to the documentation?

**Comment by BeRT2me** [comment]
How `Series.str` works for *anything* (`list`, `tuple`, `dict`, `set`, etc.) that isn't a string hasn't made much sense logically for a while. 

In my opinion, mirroring how `dict.get` does things by allowing a default value to be specified if the key/index isn't found would be a positive change. 

Not forgetting to update the documentation to match would be nice as well...

**Comment by xr-chen** [comment]
I think if we have a list of dictionaries like
 ```
lst_of_dict = [
    {"name": "Hello", "value": "World"},
    {"name": "Goodbye", "value": "Planet"}
]
```
we could construct a `DataFrame` by `pd.DataFrame(lst_of_dict)`, and `DataFrame` supports pass default value to `DataFrame.get`. I have no opinion about whether we should deprecate pass non-integer type in `Series.str.get` or not.

**Comment by xr-chen** [comment]
Should `_str_get` supports default value argument as @BeRT2me suggested?

**Comment by mroeschke** [comment]
> Should `_str_get` supports default value argument as @BeRT2me suggested?

I would save that for another issue to discuss

**Comment by mroeschke** [comment]
Appears some doc checks are failing

**Comment by xr-chen** [comment]
Honestly, I don't know what's going wrong from the Traceback of the failed check.

**Comment by mroeschke** [comment]
Might want to merge in the main branch into this PR. I believe there was some doc failures that were fixed recently.

File: pandas/tests/strings/test_strings.py [hunk 2]
```diff
@@ -828,3 +828,13 @@ def test_zfill_with_leading_sign():
     value = Series(["-cat", "-1", "+dog"])
     expected = Series(["-0cat", "-0001", "+0dog"])
     tm.assert_series_equal(value.str.zfill(5), expected)
+
+
+def test_get_with_dict_label():
+    # GH47911
+    s = Series(
+        [{"name": "Hello", "value": "World"}, {"name": "Goodbye", "value": "Planet"}]
```
Comment (mroeschke) [inline pandas/tests/strings/test_strings.py:836]: Could you also add an element where `"name"` does not exist in the dictionary?

**Review by mroeschke — APPROVED** [review]
LGTM. @phofl if you have any comments, especially opinions on supporting this.

**Comment by mroeschke** [comment]
Thanks @xr-chen

**Pull request merged**